libvncclient/rfbproto.c

```diff
@@ -1789,6 +1868,89 @@ SendClientCutText(rfbClient* client, char *str, int len)
 
 
 
+/*
+ * process server clipboard extend text
+ */
+
+static rfbBool
+rfbClientProcessExtServerCutText(rfbClient* client, char *data, int len)
+{
+  uint32_t flags;
+  if (len < sizeof(flags)) {
+    rfbClientLog("rfbClientProcessExtServerCutText. len < 4\n");
+    return FALSE;
+  }
+  memcpy(&flags, data, sizeof(flags));
+  data += sizeof(flags);
+  len -= sizeof(flags);
+  flags = rfbClientSwap32IfLE(flags);
+
+  /*only process (text | provide). Ignore all others*/
+  if (!(flags & rfbExtendedClipboard_Text)) {
+    rfbClientLog("rfbClientProcessExtServerCutText. not text type. ignore\n");
+    return TRUE;
+  }
+  if (!(flags & rfbExtendedClipboard_Provide)) {
+    rfbClientLog("rfbClientProcessExtServerCutText. not provide type. ignore\n");
+    return TRUE;
+  }
+  if (flags & rfbExtendedClipboard_Caps) {
+    rfbClientLog("rfbClientProcessExtServerCutText. caps. ignreo\n");
```

Comment:
Minor typo here, but why bail out if the caps flag is set anyway? Can't we just ignore it?

Reply:
because we just support text with notify/provide. we dont need to handle caps and this make code simple(ignore the caps handle means use the default conf). the spec make a bad decision(i think) to make all these types in one feature. maybe someday, people do copy rtf/file in/out, we can handle it here.